File: inventory-service/src/main/java/com/eiddev/shopping/inventoryservice/controller/InventoryController.java
package com.eiddev.shopping.inventoryservice.controller;

import com.eiddev.shopping.inventoryservice.model.Inventory;
import com.eiddev.shopping.inventoryservice.repository.InventoryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/inventory")
@RequiredArgsConstructor
class InventoryRestController {

    private final InventoryRepository inventoryRepository;

    @GetMapping("/{skuCode}")
    Boolean isInStock(@PathVariable String skuCode) {
        Inventory inventory = inventoryRepository.findBySkuCode(skuCode)
                .orElseThrow(() -> new RuntimeException("Cannot find product by sku code " + skuCode));
        return inventory.getStock() > 0;
    }
}

File: inventory-service-kt/src/main/kotlin/com/eiddev/shopping/inventoryservice/controller/InventoryController.kt
package com.eiddev.shopping.inventoryservice.controller

import com.eiddev.shopping.inventoryservice.model.Inventory
import com.eiddev.shopping.inventoryservice.repository.InventoryRepository
import lombok.RequiredArgsConstructor
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/inventory")
@RequiredArgsConstructor
class InventoryController (val inventoryRepository: InventoryRepository) {
    @GetMapping("/{skuCode}")
    fun isInStock(@PathVariable skuCode:String): Boolean {
        val inventory: Inventory = inventoryRepository.findBySkuCode(skuCode).orElseThrow {RuntimeException("Cannot find product by sku code $skuCode")}
        return inventory.stock > 0;
    }
}

